src/design-patterns/MazeGame/maze_factory.cpp

```diff
@@ -0,0 +1,21 @@
+#include "maze_factory.hpp"
+
+#include <iostream>
+
+MazeFactory::MazeFactory() { std::cout << "Init MazeFactory()" << std::endl; }
+
+std::unique_ptr<Maze> MazeFactory::MakeMaze() const {
+  return std::make_unique<Maze>();
+}
```

Comment:
2번째 방법

```suggestion
Maze* MazeFactory::MakeMaze() const {
  return new Maze ();
}
```

```
MazeGame 에 아래와 같이 변수가 있다고 하고, 

std::vector<std::unique_ptr <Maze>> _mazes;
MazeFactory _mfac; 

CreateMaze () 함수에서 아래와 같이 부르면, 

_mazes.emplace_back (_mfac.MakeMaze ()); 

unique_ptr로 감싸져서 _mazes 벡터에 삽입됨. 
```

std::make_unique<Maze>() 사용하는 방안이 C++14 이후에 나온 좀 더 안전한 방법이긴 함. 근데 추후에 어떻게 사용될지 모르니까, 확장성을 위해서 두가지 방법 모두 알아두는게 좋을 듯함

Reply:
서제스쳔 어케 한겨

Reply:
ㅋㅋㅋㅋcommand + g

Reply:
오 대박

Reply:
vector 없이 구현하려니까, delete 하기 귀찮아서 pass

Reply:
오케이